### ISSUE TITLE
World Breaker in Hammer mode sometimes drop instead tons of items&ores instead of matter cluster

### ISSUE THREAD

**Issue by Chemi4001** [Morpheus1101/Avaritia#77]
I hope the title is self-explanatory.
Configs I use :

```
~1.0

#General configuration of Avaritia components.
"general" {
	B:"use_end_stone"=true

	B:"break_bedrock"=true

	B:"boring_food"=false

	B:"fractured_ores"=false

	S:"aoe_trash_defaults" <
		dirt
		sand
		gravel
		netherrack
		stoneGranite
		stoneDiorite
		stoneAndesite
	>

	B:"verbose_craft_tweaker_logging"=false
}

#Disable to stop using that material in recipes. Useful if a mod adds unobtainable placeholder ores.
"materials" {
	B:"copper"=true

	B:"tin"=true

	B:"silver"=true

	B:"lead"=true

	B:"nickel"=true

	B:"steel"=true

	B:"fluxed"=true

	B:"enderium"=true

	B:"dark_steel"=false

	B:"platinum"=false

	B:"iridium"=false
}

#Balance modifications for the Compressor.
"balance" {
	I:"cost_modifier"=22

	I:"cost_multiplier"=1
}
```

Note that I removed cobblestone from aoe_trash_default, so that I can use those cobblestones for AE2 singularity in my own personal modpack. However, on the first time I use this mod (w/o modifying config file), when I tried the WB Hammer mode on stone, everything (big cube, except those in the blacklist) was reduced into items & blocks instead of matter cluster.

**Comment by Morpheus1101** [comment]
I tested the Wb in my test world an could not reproduce this issue, i consistently got matter clusters as expected.

Can you tell me the forge, Avaritia and codechickenlib version you are running

**Comment by covers1624** [comment]
Are you on a server with Sponge installed by any chance?

**Comment by Chemi4001** [comment]
Ouch, sorry. Github notification doesn't work as I expected.
I'm on MC 1.12.2.
Here's the list of mod I used

AE S2 RV5 Stable 4
Avaritia 1.12 3.2.0.13 *
Cavern 1.10.3
ChickenChunks 2.4.1.71
CCLib (codechicken) 1.12.2 3.1.5.331 *
COFHCore 4.3.10.4
COFHWorld 1.1.1.12
CT2 4.0.12
Forgelin 1.6.0
Hammercore 1.9.6.2
JEI 4.8.5.138
Magneticraft 2.0.5
MobBattlemod 2.02[1.12]
Modelloader 1.0.5
optifine HD U C6
Redst. Arsenal 2.3.10.4
Redst.Flux 2.0.1.2
RopeBridge 2.0.7
SimpleFluidTanks 1.7.1.0
SimpleQuarry 1.0.8
TherDynam 2.3.10.4
TherExpan 5.3.10.12
TherFound 2.3.10.6
WorldEdit 6.1.8

WITH
Forge1.12.2  14.23.1.2555 *

**Comment by Chemi4001** [comment]
Nope, NOT even on LAN and no sponge.
I'm a lonely guy.

**Comment by Morpheus1101** [comment]
This is an issue with HammerCore, without the mod it does not occur, i suggest to go bug them

**Comment by Chemi4001** [comment]
oh , bye bye quarries!
Guess I'll have to mine everything by myself X_X

BTW thank you for sparing your time testing.

**Comment by Morpheus1101** [comment]
If your after a really good quarry, i suggest using rftools Builder with Quarry Cards

**Comment by Chemi4001** [comment]
Thanks for your suggestion.

I'm a hardcore man so....
I think I can mine everything by myself in my hardcore world. (insert masochistic laugh here)

(I'll use CT2 to add singularity duplication recipe using AE2 Singularity(configured to require 524,288 items to create one. Some singularity will require 2 AE2 singularities to duplicate.).

**Comment by Chemi4001** [comment]
Well,
I've removed HammerCore and the frequency that the WBHammer drops gigatons(literally) of ores & stuff DRASTICALLY reduced, **_albeit not to 0%._**

However, I was "able" to pinpoint this issue a little bit more. Until now I'm not able to reproduce gigadrops on the overworld. However, this happens around 5-10% in Cavern Dimension (I haven't been able to pinpoint anything yet at that time.) and around 10-20% in Nether. I realized that sometimes when I'm breaking 2 blocks at once, WBHammer will instead drop gigatons of stuff. I then tried to mine things in a _short left click_ and then the issue **NEVER** happened (according to my 40-50 times of testing).

So in short, **mining too quick** (i.e. Like having _Efficiency V diamond pickaxe_ mining _netherrack_) can cause WBHammer to drop the lolomgwtf amount of stuff instead of matter cluster.

**Comment by Chemi4001** [comment]
Umm... excuse me?
I'm not sure if I should do this, but....

**Comment by Morpheus1101** [comment]
This issue has been tested an re-tested with a very wide range o fmods and was not able to be re-produced without HammerCore ... IF you can reproduce it 100% of the time and can narrow it down to a specific mod thats causing it, then we can re-visit this, but presently everything Works As Intended

**Comment by Chemi4001** [comment]
(Note: I went on as a staff of field trip during that time)
Here , I re-conducted the test, 100% without hammercore. To test out if it's really because of mining block too fast, I run the test in Nether (Netherrack breaks faster than stone) with Efficiency X WBH
and then rapidly click on those netherrack
![2018-02-04_11 07 09](https://user-images.githubusercontent.com/8602514/35774333-b327bb96-099e-11e8-8952-75e5f4c64e76.png)
This
![2018-02-04_11 07 17](https://user-images.githubusercontent.com/8602514/35774348-3fc15b2a-099f-11e8-91b0-d5ab8d053bd1.png)
And sometimes this
![2018-02-04_11 07 19](https://user-images.githubusercontent.com/8602514/35774351-46aa2778-099f-11e8-9b8a-59e8c3047d79.png)
I'm not sure if it has something to do with fluid also....

**Comment by Morpheus1101** [comment]
Fortune & Efficency X are not default enchantments for Minecraft and outside the scope of what would ever be possible.  If you can re-produce this normally without those 2 enchantment levels then plz let us know.

**Comment by Chemi4001** [comment]
Errmmm.... Fortune X is already in world breaker.
This time, I ran the test with normal HMWB without any special enchantements added into it in nether, but then there are still occations in that the tons of items is dropped instead. I then give myself mining fatigue and then the HMWB drops matter cluster everytime.

Also, my guess is that I think it has something to do with fluid AND too fast mining, but I'm highly unsure about this one.

**Comment by Morpheus1101** [comment]
Noone else has had this issue and this issue cannot be reproduced on its own, nor have i been able to reproduce it myself in any instance. You will need to find what mod is causing this

**Comment by Chemi4001** [comment]
Oh dear,
Now, with only 4 mods left, namely
Avaritia-1.12-3.2.0.13-universal
CodeChickenLib-1.12.2-3.1.5.331-universal
jei_1.12.2-4.8.5.138
OptiFine_1.12.2_HD_U_C8

Forge ver: 1.12.2-forge1.12.2-14.23.1.2555
Java ver: jre1.8.0_161

And the issue still persist for me.....

**Comment by covers1624** [comment]
Are you on sponge? That is literelly the only thing that could cause this.

**Comment by Chemi4001** [comment]
Ahh, I see why now.
It's because I've created my own mod 3 years ago and hide the jar file (wtf was I thinking???) and what it does is modifying the block breaking event so that the mining speed become "more" realistic.

This issue can finally be closed now.
